### ISSUE TITLE
Bypassed request's referrerPolicy is reset to default

### ISSUE THREAD

**Issue by ne2030** [kettanaito/msw#1380]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Chromium (Chrome, Brave, etc.)

### Reproduction repository

https://github.com/ne2030/msw-issue

### Reproduction steps

1. Start a local server at root directory. For example, if you have Python on your machine you could start a server running on port 8001 using `python -m SimpleHTTPServer 8001` for Python 2.x, or `python3 -m http.server 8001` for Python 3.x.
2. Navigate to the local server, e.g. [http://localhost:8001/](http://localhost:8001/)
3. Open the chrome dev tool - network tab and check whether the problem is reproduced.

Reproduction repository does not use MSW directly, instead, focuses on Fetch issue. 

### Current behavior

Referrer policy is reset to default 'strict-origin-when-cross-origin' when the request is fetched with cloned request and custom options. 

```javascript
const headers = Object.fromEntries(clonedRequest.headers.entries())
delete headers['x-msw-bypass']

return fetch(clonedRequest, { headers })
```

Here, if any options are passed as a second argument to fetch API, then the referrer policy is reset! I don't know the exact cause of this problem, but it can be solved below.

```javascript
const headers = Object.fromEntries(clonedRequest.headers.entries())

delete headers['x-msw-bypass']

return fetch(clonedRequest, { headers, referrerPolicy: request.referrerPolicy, referrer: request.referrer })
```

Or, I saw msw condesandbox and there, this problem is not reproduced and found out mockserviceworker.js file is different from deployed version (0.45.0). I think the version on condsandbox that creates a new Request instance and uses this directly as argument with no options at Fetch can also handle this problem. 

```javascript
function serializeHeaders(headers) {
  const reqHeaders = {};
  headers.forEach((value, name) => {
    reqHeaders[name] = reqHeaders[name]
      ? [].concat(reqHeaders[name]).concat(value)
      : value;
  });
  return reqHeaders;
}

const cleanRequestHeaders = serializeHeaders(clonedRequest.headers);

// Remove the bypass header to comply with the CORS preflight check.

const originalRequest = new Request(clonedRequest, {
  headers: new Headers(cleanRequestHeaders)
});

fetch(originalRequest);
```

The below screenshots are for reference.

<img width="988" alt="스크린샷 2022-08-30 오후 12 21 42" src="https://user-images.githubusercontent.com/16914062/187359105-dcbf870e-7d0c-44b3-a6ea-5d509ddb98fd.png">
<img width="781" alt="스크린샷 2022-08-30 오후 12 21 38" src="https://user-images.githubusercontent.com/16914062/187359137-529cd05d-dee3-4935-904e-a1a4037f582c.png">
<img width="920" alt="스크린샷 2022-08-30 오후 12 21 33" src="https://user-images.githubusercontent.com/16914062/187359147-39ef225b-8835-49bd-8e0f-e7f35da353f9.png">

### Expected behavior

Preserve original option of request (especially referrerPolicy and referrer)

**Comment by ne2030** [comment]
Is there any updates on this?

**Comment by kettanaito** [comment]
## Released: v2.0.0 🎉

This has been released in v2.0.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.0.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.0.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
